file/builder.go

```diff
@@ -56,9 +56,11 @@ func (b *stateBuilder) build() (*utils.KongRawState, *utils.KonnectRawState, err
 	if b.targetContent.Info != nil {
 		kongDefaults = b.targetContent.Info.Defaults
 	}
-	b.defaulter, err = defaulter(kongDefaults)
-	if err != nil {
-		return nil, nil, fmt.Errorf("creating defaulter: %w", err)
+	if !isKongDefaultEmpty(kongDefaults) || b.defaulter == nil {
+		b.defaulter, err = defaulter(kongDefaults)
+		if err != nil {
+			return nil, nil, fmt.Errorf("creating defaulter: %w", err)
+		}
```

Comment:
The KongDefaults thing hasn't been around for long enough and most users won't fill it in.
We should have a precedence order here:
- Take an entity, populate fields take first priority
- Fill defaults as present in KongDefaults
- Fill defaults via the API

Reply:
A couple of comments after my last commit:
- this change makes the hardcoded defaults in `deck` obsolete, so I'm removing them here
- which means I had to rework a bit the tests too. I moved the "old" default objects inside the test files in order to minimize the changes on tests, since we are already changing a lot of code. This should keep confidence of changes higher

let me know what you think!